packages/components/src/components/Drawer/Drawer.tsx

```diff
@@ -12,32 +12,53 @@ export interface DrawerProps extends DialogProps {
 const StyledBackdrop = styled(Box.div)`
   background-color: rgba(39, 49, 63, 0.5);
   backdrop-filter: blur(3px);
+  opacity: 0;
+  transition: all 150ms;
+
+  &[data-enter] {
+    backdrop-filter: blur(3px);
+    opacity: 1;
+  }
+`;
+
+const StyledDrawer = styled(Box.div)`
+  background-color: ${theme.colors.colorBackground};
+  box-shadow: ${theme.shadows.shadowStrong};
+  display: flex;
+  flex-direction: column;
+  max-width: 34.375rem;
+  position: fixed;
+  right: calc(-100%);
+  top: 0;
+  bottom: 0;
+  transition: right 0.6s cubic-bezier(0.4, 0, 0.2, 1);
+  z-index: ${theme.zIndices.zIndex30};
+
+  &[data-enter] {
+    right: 0;
+  }
 `;
 
 /** A Drawer is a page overlay that displays information and blocks interaction with the page until an action is taken or the Drawer is dismissed. */
 const Drawer = React.forwardRef<HTMLDivElement, DrawerProps>(
-  ({ children, initialFocusRef, ...props }, ref) => {
+  ({ children, state, initialFocusRef, ...props }, ref) => {
+    const internalState = useDialogState({
+      ...state,
+      animated: true,
```

Comment:
This will no longer be necessary when we update ariakit to new version as described here https://newsletter.ariakit.org/p/animation-ariakit-plus-and-more?utm_source=profile&utm_medium=reader2.